### PR TITLE
[SPARK-49527] Add `ConfOptionDocGenerator` to generate Spark Operator Config Property Doc

### PR DESCRIPTION
**Pull request by jiangzho**
### What changes were proposed in this pull request?

This PR adds a module `docs-utils` to automatically generate config properties doc page from source.

### Why are the changes needed?

This helps to keep config property docs up-to-date for properties by adding it as a gradle task.

### Does this PR introduce _any_ user-facing change?

No (doc only, not released)

### How was this patch tested?

Pass the CIs

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by jiangzho** [comment]
also cc @viirya

File: docs-utils/build.gradle [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ext {
+    javaMainClass = "org.apache.spark.k8s.operator.utils.ConfOptionDocGenerator"
```
Comment (dongjoon-hyun) [inline docs-utils/build.gradle:21]: 2-space indentation please.

**Review by dongjoon-hyun — COMMENTED** [review]
Was it technically impossible to put under `build-tools` as I suggested?

- https://github.com/apache/spark-kubernetes-operator/pull/113#discussion_r1744847803

File: settings.gradle [hunk 2]
```diff
@@ -20,3 +20,5 @@ rootProject.name = 'apache-spark-kubernetes-operator'
 include 'spark-operator-api'
 include 'spark-submission-worker'
 include 'spark-operator'
+include 'docs-utils'
+
```
Comment (dongjoon-hyun) [inline settings.gradle:24]: Please remove extra line.

**Review by dongjoon-hyun — COMMENTED** [review]
Could you make CIs happy?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @jiangzho .
Merged to main.

**Comment by dongjoon-hyun** [comment]
Let me re-trigger it.

**Comment by dongjoon-hyun** [comment]
Merged to main.

**Pull request closed without merging**